libc/stdlib_alloc.c

```diff
@@ -0,0 +1,561 @@
+{
+	int NUMBERS[BLOCK_SIZE];
+	int *ptr = calloc(1, sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr);
+	TEST_ASSERT_EQUAL_INT(0, *ptr);
+	ptr[0] = 1;
+
+	int *ptr1 = realloc(ptr, 2 * sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr1);
+	TEST_ASSERT_EQUAL_INT(0, (uintptr_t)ptr1 % sizeof(int));
+	TEST_ASSERT_EQUAL_INT(1, ptr1[0]);
+
+	int *ptr2 = calloc(BLOCK_SIZE / 8, BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr2);
+	TEST_ASSERT_EACH_EQUAL_INT(0, ptr2, BLOCK_SIZE);
+	for (int i = 0; i < BLOCK_SIZE; i++) {
+		ptr2[i] = i;
+		NUMBERS[i] = i;
+	}
+	TEST_ASSERT_EQUAL_INT_ARRAY(NUMBERS, ptr2, BLOCK_SIZE);
+
+	int *ptr3 = realloc(ptr2, 2 * (BLOCK_SIZE * LARGE_SIZE));
+	TEST_ASSERT_NOT_NULL(ptr3);
+	TEST_ASSERT_EQUAL_INT_ARRAY(NUMBERS, ptr3, BLOCK_SIZE);
+
+	free(ptr1);
+	free(ptr3);
+}
+
+
+TEST(stdlib_malloc, realloc_malloc_resize)
+{
+	int *ptr = malloc(sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr);
+	memset(ptr, 0x40, sizeof(int));
+	TEST_ASSERT_EQUAL_INT(0, (uintptr_t)ptr % sizeof(int));
+
+	int *ptr1 = realloc(ptr, BLOCK_SIZE * sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr1);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x40, ptr1, sizeof(int));
+	TEST_ASSERT_EQUAL_INT(0, (uintptr_t)ptr1 % sizeof(int));
+
+	free(ptr1);
+}
+
+
+TEST(stdlib_malloc, realloc_malloc_resize_smaller)
+{
+	int *ptr = malloc(2 * sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr);
+	memset(ptr, 0x40, 2 * sizeof(int));
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x40, ptr, 2 * sizeof(int));
+
+	int *ptr1 = realloc(ptr, sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr1);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x40, ptr1, sizeof(int));
+	TEST_ASSERT_EQUAL_INT(0, (uintptr_t)ptr1 % sizeof(int));
+
+	int *ptr2 = malloc(2 * (BLOCK_SIZE * LARGE_SIZE));
+	TEST_ASSERT_NOT_NULL(ptr2);
+	memset(ptr2, 0x80, 2 * (BLOCK_SIZE * LARGE_SIZE));
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x80, ptr2, 2 * (BLOCK_SIZE * LARGE_SIZE));
+
+	int *ptr3 = realloc(ptr2, BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr3);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x80, ptr3, BLOCK_SIZE * LARGE_SIZE);
+
+	free(ptr1);
+	free(ptr3);
+}
+
+
+TEST(stdlib_malloc, realloc_malloc_resize_larger)
+{
+	int *ptr = malloc(sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr);
+	memset(ptr, 0x40, sizeof(int));
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x40, ptr, sizeof(int));
+
+	int *ptr1 = realloc(ptr, 2 * sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr1);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x40, ptr1, sizeof(int));
+	TEST_ASSERT_EQUAL_INT(0, (uintptr_t)ptr1 % sizeof(int));
+
+	int *ptr2 = malloc(BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr2);
+	memset(ptr2, 0x80, BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x80, ptr2, BLOCK_SIZE * LARGE_SIZE);
+
+	int *ptr3 = realloc(ptr2, 2 * (BLOCK_SIZE * LARGE_SIZE));
+	TEST_ASSERT_NOT_NULL(ptr3);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x80, ptr3, (BLOCK_SIZE * LARGE_SIZE));
+
+	free(ptr1);
+	free(ptr3);
+}
+
+
+TEST(stdlib_malloc, realloc_multiple)
+{
+	int *ptr = calloc(BLOCK_SIZE, BLOCK_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr);
+	TEST_ASSERT_EACH_EQUAL_INT(0, ptr, BLOCK_SIZE);
```

Comment:
You only have checked `sizeof(int) * BLOCK_SIZE` bytes, we allocated `BLOCK_SIZE * BLOCK_SIZE` bytes. Fix the sizes or use char \*/unsigned char\*